types/underscore/index.d.ts

```diff
@@ -998,26 +1003,32 @@ declare module _ {
         ): number;
 
         /**
-        * A function to create flexibly-numbered lists of integers, handy for each and map loops. start, if omitted,
-        * defaults to 0; step defaults to 1. Returns a list of integers from start to stop, incremented (or decremented)
-        * by step, exclusive.
-        * @param start Start here.
-        * @param stop Stop here.
-        * @param step The number to count up by each iteration, optional, default = 1.
-        * @return Array of numbers from `start` to `stop` with increments of `step`.
-        **/
-
+         * A function to create flexibly-numbered lists of integers, handy for
+         * `each` and `map` loops. Returns a list of integers from `start`
+         * (inclusive) to `stop` (exclusive), incremented (or decremented) by
+         * `step`, exclusive. Note that ranges that `stop` before they `start`
+         * are considered to be zero-length instead of negative - if you'd like
+         * a negative range, use a negative `step`.
+         * @param start The number to start at, optional, default = 1.
+         * @param stop The number to stop at.
+         * @param step The number to count up by each iteration, optional,
+         * default = 1.
+         * @returns An array of numbers from `start` to `stop` with increments
+         * of `step`.
+         **/
         range(
             start: number,
             stop: number,
-            step?: number): number[];
+            step?: number
+        ): number[];
 
         /**
-        * @see _.range
-        * @param stop Stop here.
-        * @return Array of numbers from 0 to `stop` with increments of 1.
-        * @note If start is not specified the implementation will never pull the step (step = arguments[2] || 0)
-        **/
+         * A function to create flexibly-numbered lists of integers, handy for
+         * `each` and `map` loops.  Returns a list of integers from 0
+         * (inclusive) to `stop` (exclusive), incremented by 1.
+         * @param stop The number to stop at.
+         * @returns An array of numbers from 0 to `stop` with increments of 1.
+         **/
         range(stop: number): number[];
```

Comment:
The issue there is that this doesn't use the same parameter name in this position in both cases. For that to work, I'd have to call this `startOrStop` and the next optional parameter would be `stop`, and I feel like documentation might get confusing in that case. I'll do it, though, if you prefer.

Reply:
Do's and Don'ts is definitely clear on parameters with the same name being collapsed, but I feel like it doesn't definitively cover the case where the parameters serve different purposes and thus have different names in each overload.

Reply:
I suppose it does discuss how one could potentially pass `undefined` as the second parameter and not doing that does potentially block the effective use of something like a tuple spread...but on the other hand the purpose of the item in the first position of said theoretical tuple spread changes significantly depending on what ends up in the second position, so I'm not really sure how realistic a scenario that is.

Reply:
I'm not entirely sure because I'm not following you about the tuple spread, but assuming that that will not cause me to change my mind, I think I prefer the unified signature.

Reply:
Oh, interesting, with the current signatures the seemingly valid argument set `_.range(10, undefined, 2)`, which results in `[0, 2, 4, 6, 8]`, can't be used.

Reply:
Updated and added tests for the stop + step parameter set described above.